modules/light-clients/08-wasm/keeper/grpc_query.go

```diff
@@ -25,17 +22,17 @@ func (k Keeper) Code(c context.Context, req *types.QueryCodeRequest) (*types.Que
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
-	ctx := sdk.UnwrapSDKContext(c)
-	store := ctx.KVStore(k.storeKey)
-
 	codeHash, err := hex.DecodeString(req.CodeHash)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, "invalid code hash")
 	}
 
-	codeKey := types.CodeHashKey(codeHash)
-	code := store.Get(codeKey)
-	if code == nil {
+	// Only return code hashes we previously stored, not arbitrary code hashes that might be stored via e.g Wasmd.
+	if !types.HasCodeHash(sdk.UnwrapSDKContext(c), k.cdc, codeHash) {
```

Comment:
[nit] c -> goCtx for consistency (I know this variable wasn't introduced in this PR, sorry!)